src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Validation/IAsyncValidatableObject.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Validation
```

Comment:
I'd put the whole thing into `OrchardCore.Mvc.Core`, and use namespaces reflecting the folder structure.